scripts/process/as_svg.R

```diff
@@ -0,0 +1,266 @@
+  }, USE.NAMES = FALSE)
+}
+
+#' sub text and/or define new svg attributes
+#' 
+#' uses `sub_attribute_text` to create attribute strings
+#' 
+#' @param dataframe data to use for evaluation
+#' @param attributes key value pairs that may or may not contain 
+#' mustache {{key}} syntax
+#' @return a data.frame with new columns for each name in `attributes`
+svg_sub_attributes <- function(dataframe, attributes = NULL){
+  
+  if (!is.null(attributes)){
+    # replace existing attributes w/ these user-specified ones
+    # evaluate "mustache" keys w/ geom_df
+    attr_names <- names(attributes)
+    
+    # if mustache keys exist, evaluate with the context of the dataframe variables:
+    attrs_subbed <- sapply(attr_names, FUN = function(x) {
+      sub_attribute_text(template = attributes[[x]], data = dataframe)
+    })
+    
+    dataframe[attr_names] <- attrs_subbed
+  }
+  
+  return(dataframe)
+}
+
+#' convert a simple feature collection into an svg path element's `d` attribute
+#' 
+#' @param .object a simple feature collection, currently limited to polygons
+#' @param xlim min and max x values in the plotting range in the coordinates of `sfc_object`
+#' @param ylim min and max y values in the plotting range in the coordinates of `sfc_object`
+#' @param \dots additional arguments sent to `plot`
+#' @param width the width of the output figure in inches
+#' @param height the height of the output figure in inches
+#' @param pointsize used in svgstring...fonts?
+#' 
+#' @return a data.frame with the same data as `.object`, 
+#' but with the new attribute(s) added, and the geometry removed
+sf_to_path <- function(.object, xlim, ylim, ..., width, height, pointsize = 12){
+  
+  stopifnot(!is.null(.object))
+  stopifnot(packageVersion('svglite') == '1.2.0.9003')
+  
+  sfc_object <- .object[["geometry"]]
+  data_in <- data.frame(.object) %>% select(-geometry)
+  data_in$feature_id <- seq_len(nrow(data_in))
+  
+  # hmmm...with sf, each polygon of a multi-poly plots as a separate `d`. 
+  # Can get this w/ st_cast(sfc_object[j], "POLYGON")
+  # could test for this `"sfc_MULTIPOLYGON" %in% class(sfc_object[j])`
+  # use `feature_id` to track which feature each polygon came from
+  feature_ids <- c()
+  rendered <- svglite::xmlSVG(width = width, height = height, pointsize = pointsize, standalone = F, {
+    
+    set_svg_plot(xlim, ylim)
+    for (j in seq_len(length(sfc_object))){
+      for (poly in sf::st_cast(sfc_object[j], "POLYGON")){
+        plot(poly, ..., add = TRUE)
+        feature_ids <- c(feature_ids, j)
+      }
+    }
+  })
+  
+  
+  paths <- xml_child(rendered) %>% xml_children
+
+  # collapse the multi-d into a single one: 
+  data_out <- data.frame(d_raw = xml_attr(paths, 'd'), 
+                         feature_id = feature_ids, 
+                         stringsAsFactors = FALSE) %>% 
+    group_by(feature_id) %>% 
+    summarize(d = paste(d_raw, collapse = ' ')) %>% 
+    left_join(data_in) %>% select(-feature_id)
```

Comment:
Can you include `dplyr::` for each of these functions? Or use `@importFrom`/`@import`? You have a mix of including or excluding explicit package calls right now.